addons/mail/models/mail_template.py

```diff
@@ -176,6 +179,11 @@ def generate_email(self, res_ids, fields):
             # update values for all res_ids
             for res_id in template_res_ids:
                 values = results[res_id]
+                if  'author' in fields and values.get('author'):
+                    self.author_id = self.env['res.users'].browse(int(values.get('author')))
```

Comment:
Why writing on self.author_id ? If we loop on 200 records, we may have 200 different values ... directly setting author_id and email_from in values should be sufficent.

I would also not override email_from is already computed / set (we might want to set an author and another email_from).

Reply:
Correct me if I'm wrong but here I set email_from only if there it was requested and there is no value set currently. So it will not be override if we already have a value for the email.